### PR TITLE
8261221: Tooltip bigger than screen size blinks - shows and hides over and over again

### PR DESCRIPTION
**Pull request by xardif**
`Tooltip` is no longer hiding upon receiving `MouseEvent.MOUSE_ENTERED_TARGET` event inside it. Pressing mouse on overlaying tooltip also kills the tooltip, so the infinite duration tooltip can be closed.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [ ] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8261221](https://bugs.openjdk.org/browse/JDK-8261221): Tooltip bigger than screen size blinks - shows and hides over and over again

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx.git pull/395/head:pull/395` \
`$ git checkout pull/395`

Update a local copy of the PR: \
`$ git checkout pull/395` \
`$ git pull https://git.openjdk.org/jfx.git pull/395/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 395`

View PR using the GUI difftool: \
`$ git pr show -t 395`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/395.diff">https://git.openjdk.org/jfx/pull/395.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jfx/pull/395#issuecomment-780742024)

### CODE REVIEW

File: modules/javafx.controls/src/main/java/javafx/scene/control/Tooltip.java [hunk 1]
```diff
@@ -1028,6 +1031,11 @@ public boolean isSettable(Tooltip.CSSBridge cssBridge) {
          * stop the HIDE_TIMER, hide the tooltip, and start the LEFT_TIMER.
          */
         private EventHandler<MouseEvent> LEAVING_HANDLER = (MouseEvent event) -> {
+            if( mouseInsideTooltip )
```
Comment (kevinrushforth) [inline modules/javafx.controls/src/main/java/javafx/scene/control/Tooltip.java:1034]: Can you reformat this to follow our code style guidelines? Add a space after the `if`, remove the spaces after `(` and before `)`, and move the opening `{` to the same line as the `if`.

**Review by kevinrushforth — COMMENTED** [review]
I'll review and test it later. This will need a unit test (or system test if a unit test is infeasible). I also left a minor formatting comment below.

File: modules/javafx.controls/src/main/java/javafx/scene/control/Tooltip.java [hunk 2]
```diff
@@ -1074,6 +1092,9 @@ private void install(Node node, Tooltip t) {
             node.addEventHandler(MouseEvent.MOUSE_MOVED, MOVE_HANDLER);
             node.addEventHandler(MouseEvent.MOUSE_EXITED, LEAVING_HANDLER);
             node.addEventHandler(MouseEvent.MOUSE_PRESSED, KILL_HANDLER);
+            t.addEventFilter( MouseEvent.MOUSE_ENTERED_TARGET, TOOLTIP_MOUSE_ENTERED );
```
Comment (kevinrushforth) [inline modules/javafx.controls/src/main/java/javafx/scene/control/Tooltip.java:1095]: Code style: remove spaces after `(` and before `)` here and similar method calls elsewhere.

**Comment by kevinrushforth** [comment]
One more request: Please follow the instructions [here](https://github.com/openjdk/jfx/pull/395/checks) to enable running pre-submit tests.

File: modules/javafx.controls/src/main/java/javafx/scene/control/Tooltip.java [hunk 3]
```diff
@@ -859,6 +859,7 @@ public boolean isSettable(Tooltip.CSSBridge cssBridge) {
 
         private boolean hideOnExit;
         private boolean cssForced = false;
+        private boolean mouseInsideTooltip = true;
```
Comment (aghaisas) [inline modules/javafx.controls/src/main/java/javafx/scene/control/Tooltip.java:862]: Default value of this member should be false.
This boolean indicates a state where mouse is inside a Tooltip - it cannot be assumed true to begin with.

**Review by kevinrushforth — COMMENTED** [review]
I see that the unit tests fails on a couple platforms (although mac failure looks like a network problem with the GitHub actions server). Since you are several commits out of date, can you merge in the latest changes from upstream master into your branch?

And speaking of tests, can you create an automated unit test to verify this fix?

I presume you've tested this with normal sized tooltips. Have you found any differences in behavior in those cases with the added event filtering?

I also left one more formatting comment below.

**Comment by xardif** [comment]
@kevinrushforth sorry for "little" delay in answering, but I thing that previously failing tests have passed as successful now.

**Pull request closed without merging**